File: front-end/Norlan-Svelte-TypeScript-master/src/lib/models/CorsoFormazione.ts
import { StatoCorso } from './Enums';
/*
Modello di dominio per i corsi di formazione.
Definisce i dettagli logistici del corso, lo stato attuale e i riferimenti al docente.
*/
export interface CorsoData {
	idCorso: number;
	titolo: string;
	dataOrario: string;
	luogoFisico: string;
	capacitaMassima: number;
	stato: StatoCorso;
	idDocente: number;
	emailDocente: string;
}

export class CorsoFormazione {
	idCorso: number;
	titolo: string;
	dataOrario: string;
	luogoFisico: string;
	capacitaMassima: number;
	stato: StatoCorso;
	idDocente: number;
	emailDocente: string;

	constructor(data: CorsoData) {
		this.idCorso = data.idCorso;
		this.titolo = data.titolo;
		this.dataOrario = data.dataOrario;
		this.luogoFisico = data.luogoFisico;
		this.capacitaMassima = data.capacitaMassima;
		this.stato = data.stato;
		this.idDocente = data.idDocente;
		this.emailDocente = data.emailDocente;
	}
}

File: back-end/norlan-client-portal-master/src/main/java/it/norlan/clientportal/model/CorsoFormazione.java
package it.norlan.clientportal.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

/**
 * Entità JPA che modella un Corso di Formazione (es. Sicurezza sul Lavoro).
 * Gestisce le coordinate temporali e logistico-spaziali dell'evento, l'assegnazione
 * del formatore (Docente) e il ciclo di vita (Stato) del registro presenze.
 */

@Getter @Setter
@NoArgsConstructor @AllArgsConstructor
@Entity
@Table(name = "corso_formazione")
public class CorsoFormazione {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer idCorso;

    @Column(nullable = false)
    private String titolo;

    @Column(name = "data_orario", nullable = false)
    private LocalDateTime dataOrario;

    @Column(name = "luogo_fisico", nullable = false)
    private String luogoFisico;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_docente", nullable = false)
    private Docente docente;

    @Enumerated(EnumType.STRING)
    private StatoCorso stato = StatoCorso.PROGRAMMATO;

    public enum StatoCorso {
        PROGRAMMATO,
        IN_SVOLGIMENTO,
        CONCLUSO,
        ATTESA_FIRMA_DOCENTE,
        VALIDATO,
        CERTIFICATO
    }
}
